src/VecSim/algorithms/brute_force/brute_force.cpp

```diff
@@ -199,6 +202,11 @@ VecSimQueryResult_List BruteForceIndex::topKQuery(const void *queryBlob, size_t
         size_t block_size = vectorBlock->getLength();
         vecsim_stl::vector<float> scores(block_size, this->allocator);
         for (size_t i = 0; i < block_size; i++) {
+            if (__builtin_expect(VecSimIndex::timeoutCallback(timeoutCtx), 0)) {
```

Comment:
Can you explain what `__builtin_expect` does? Also, I would suggest renaming `timeoutCallback` to something like: `ReachedTimeout()`